presto-orc/src/main/java/com/facebook/presto/orc/OrcOutputBuffer.java

```diff
@@ -386,12 +388,14 @@ public String toString()
 
     private void ensureWritableBytes(int minWritableBytes)
     {
+        assert slice.length() <= maxBufferSize;
+
         int neededBufferSize = bufferPosition + minWritableBytes;
         if (neededBufferSize <= slice.length()) {
             return;
         }
 
-        if (slice.length() >= maxBufferSize) {
+        if (slice.length() == maxBufferSize) {
```

Comment:
as defensive coding, this should be still >= and let the assert inside writeChunkToOutputStream fire.

Reply:
This statement works with the assert above (`slice.length() <= maxBufferSize`), which makes sure that `maxBufferSize > slice.length()` is never the case, and we handle the only two other cases.

But I think your suggestions also make sense.